### PR TITLE
[routing-utils] Exclude /.well-known from trailing slash redirect

### PR DESCRIPTION
**Pull request by dav-is**
`/.well-known` files shouldn't have trailing slashes added as these clients likely do not follow redirects

Specifically, `/.well-known/apple-developer-merchantid-domain-association` cannot have trailing slash.

### Related Issues

https://vercel.slack.com/archives/CLDDX2Y0G/p1605127589058800

#### Tests

- [x] The code changed/added as part of this PR has been covered with tests
- [x] All tests pass locally with `yarn test-unit`

#### Code Review

- [x] This PR has a concise title and thorough description useful to a reviewer
- [x] Issue from task tracker has a link to this PR

### CODE REVIEW

**Review by styfle — COMMENTED** [review]
Looks good but needs tests

**Comment by dav-is** [comment]
@styfle I added a test

**Comment by styfle** [comment]
@dav-is The `convertTrailingSlash enabled` test is failing

File: packages/now-routing-utils/test/superstatic.spec.js [hunk 1]
```diff
@@ -671,6 +671,7 @@ test('convertHeaders', () => {
 test('convertTrailingSlash enabled', () => {
   const actual = convertTrailingSlash(true);
   const expected = [
+    { src: '^/\\.well-known(?:/.*)?$' },
```
Comment (styfle) [inline packages/now-routing-utils/test/superstatic.spec.js:674]: You need to also update the `mustMatch` and `mustNotMatch` arrays below

**Review by styfle — APPROVED** [review]

**Review by ijjk — APPROVED** [review]

**Pull request merged**